lab-kyle/test/note-router-test.js

```diff
@@ -0,0 +1,134 @@
+request.use(superPromise);
+
+describe('testing module note-router', function(){
+  before((done) => {
+    if (!server.isRunning){
+      server.listen(port , () => {
+        server.isRunning = true;
+        console.log('server up ::', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    if (server.isRunning){
+      server.close(() => {
+        server.isRunning = false;
+        console.log('server down');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('POST /api/note with valid data', function(){
+    before((done) => {
+      userCrud.createUser({name: 'test name', email: 'test email'})
+      .then( user => {
+        this.tempUser = user;
+        done();
+      })
+      .catch(done);
+    });
+
+    after((done) => {
+      noteCrud.removeAllNotes()
+      .then( () => done())
+      .catch(done);
+    });
+
+    it('should return a note', (done) => {
+      request.post(`${baseUrl}/api/note`)
+      .send({userId: this.tempUser._id, content: 'test note', name: 'thisNote'})
+      .then( res => {
+        expect(res.status).to.equal(200);
+        expect(res.body.userId).to.equal(`${this.tempUser.id}`);
+        done();
+      })
+      .catch(done);
+    });
+  });
+
+  describe('POST /api/note with bad data', function(){
+    before((done) => {
+      userCrud.createUser({name: 'test name', email: 'test email'})
+      .then( user => {
+        this.tempUser = user;
+        done();
+      })
+      .catch(done);
+    });
+
+    after((done) => {
+      noteCrud.removeAllNotes()
+      .then( () => done())
+      .catch(done);
+    });
+
+    it('should return a note', (done) => {
+      request.post(`${baseUrl}/api/note`)
+      .send({userId: this.tempUser._id, cont: 'test note', nae: 'thisNote'})
+      .then((done))
+      .catch((err) => {
+        try {
+          var res = err.response;
+          expect(res.status).to.equal(400);
+          expect(res.text).to.equal('bad request');
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+    });
+  });
+
+  describe('GET /api/note with valid id', function(){
+    before((done) => {
+      userCrud.createUser({name: 'test name', email: 'test email'})
+      .then( user => {
+        this.tempUser = user;
+        return noteCrud.createNote({userId: this.tempUser._id, name: 'test note', content: 'this is a test note'});
+      })
+      .then( note => {
+        this.tempNote = note;
+        done();
+      })
+      .catch(done);
+    });
+
+    after((done) => {
+      userCrud.removeAllUsers();
+      noteCrud.removeAllNotes()
+      .then( () => done())
+      .catch(done);
+    });
+
+    it('should return a note', (done) => {
+      request.get(`${baseUrl}/api/note/${this.tempUser.id}`)
+      .then( res => {
+        expect(res.status).to.equal(200);
+        expect(res.body.noteId).to.equal(`${this.tempUser.id}`);
```

Comment:
This test was failing. You just need to pay attention to what your functions are returning. Firstly you're expecting a noteId to equal a userId; this doesn't make sense.
Secondly, your get request to note/:userId returns all that users notes, so you are expecting an array of notes, therefore res.body will be an array and res.body.anything is undefined. To get this test to pass you just needed to expect(res.body[0].userId).to.equal(`${this.tempUser.id}`);